### PR TITLE
Tm/upgrade-ledger-2fa

### PR DESCRIPTION
**Pull request by arhtudormorar**
#### Description of the pull request 
Changed ledger 2FA signing method from using WebWallet provider to CrossWindow provider

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [x] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md [hunk 1]
```diff
@@ -44,16 +44,27 @@ All signing providers (except Ledger) take care internally of formatting the tra
...
+import { CrossWindowProvider } from "@multiversx/sdk-web-wallet-cross-window-provider";
+
+// instantiate wallet cross-window provider
+await CrossWindowProvider.getInstance().init();
+const crossWindowProvider = CrossWindowProvider.getInstance();
+crossWindowProvider.setWalletUrl(WALLET_PROVIDER_URL);
+
+// set sender
+const ledgerSenderBech32 = await this.hwProvider.getAddress();
+const ledgerSender = new Address(senderBech32);
```
Comment (andreibancioiu) [inline docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md:56]: `Address.newFromBech32()`;

https://github.com/multiversx/mx-sdk-js-core/blob/main/src/address.ts#L95

File: docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md [hunk 2]
```diff
@@ -44,16 +44,27 @@ All signing providers (except Ledger) take care internally of formatting the tra
...
+const sender = ledgerSender; // or "erd1...abc" witohut awaiting `getAddress()`
```
Comment (andreibancioiu) [inline docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md:57]: Can we remove this variable / simplify the code?

File: docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md [hunk 3]
```diff
@@ -44,16 +44,27 @@ All signing providers (except Ledger) take care internally of formatting the tra
...
+crossWindowProvider.setAddress(sender);
+
+// the user signs transactions on ledger so we need to perform an extra
```
Comment (andreibancioiu) [inline docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md:60]: Comma after "the user signs transactions on ledger".

Comment (andreibancioiu) [reply docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md:60]: _To complete the transaction, the user will sign using their Ledger device. This requires an additional step: a confirmation popup will appear, prompting the user to approve the action, after which a new tab will open in the browser._

(or something similar)

File: docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md [hunk 4]
```diff
@@ -44,16 +44,27 @@ All signing providers (except Ledger) take care internally of formatting the tra
...
+const guardedTransactions = await crossWindowProvider.guardTransactions(
+    signedTransactions
+);
 ```
-Once transactions are back from the web wallet, they can be retrieved as follows:
 
-```js
-const signedTransactions = new WalletProvider('https://wallet.multiversx.com/dapp/init').getTransactionsFromWalletUrl();
-```
+For a working example see the code used in [signing-providers](https://github.com/multiversx/mx-sdk-js-examples/blob/594465208fd3a9d5c57bca8ecc94f2dc59cbf4a6/signing-providers/src/hw.js#L87)
```
Comment (andreibancioiu) [inline docs/sdk-and-tools/sdk-dapp/internal-processes/guardians.md:69]: Can we have a short tag, instead of a commit hash?

**Review by andreibancioiu — DISMISSED** [review]

**Review by razvantomegea — DISMISSED** [review]

**Review by andreigiura — APPROVED** [review]

**Review by razvantomegea — APPROVED** [review]

**Pull request merged**